display_list/display_list_complexity.h

```diff
@@ -0,0 +1,48 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_FLOW_DISPLAY_LIST_COMPLEXITY_H_
+#define FLUTTER_FLOW_DISPLAY_LIST_COMPLEXITY_H_
+
+#include "flutter/display_list/types.h"
+#include "third_party/skia/include/gpu/GrTypes.h"
+
+namespace flutter {
+
+class DisplayListComplexityCalculator {
+ public:
+  static DisplayListComplexityCalculator* GetForSoftware();
+  static DisplayListComplexityCalculator* GetForBackend(GrBackendApi backend);
+
+  virtual ~DisplayListComplexityCalculator() = default;
+
+  // Returns a calculated complexity score for a given DisplayList object
+  virtual int compute(DisplayList* display_list) = 0;
```

Comment:
I would change this method and op_count to unsigned. We might want 64-bit unsigned eventually, but at least uint32_t for now.